src/plugins/intel_cpu/src/nodes/subgraph.cpp

```diff
@@ -965,11 +992,34 @@ void Subgraph::SubgraphExecutor::parallel_forNd(const std::function<void(jit_sni
                 tmp /= dom[j];
             }
 
-            caller(call_args, indexes.data());
+            caller(call_args, indexes);
         }
     });
 }
 
+void Subgraph::SubgraphExecutor::execute(dnnl::stream strm, std::vector<MemoryPtr>& inMemPtrs, std::vector<MemoryPtr>& outMemPtrs) {
+    if (m_in_requested_descs.empty())
+        exec_impl(inMemPtrs, outMemPtrs);
+    else
+        reorder_execute(strm, inMemPtrs, outMemPtrs);
```

Comment:
Please see [this comment](https://github.com/openvinotoolkit/openvino/pull/27007#discussion_r1844107296): in case of reordering we must make a copy of `inMemPtrs`. So the current approach avoids unnecessary copy if reordering is not needed

Reply:
Refactored, please take a look